amd_openvx_extensions/amd_rpp/source/tensor/ColorCast.cpp

```diff
@@ -0,0 +1,287 @@
+                data->pRgbFloats[dstIdx + 1] = data->pRgbFloats[srcOff + 1];
+                data->pRgbFloats[dstIdx + 2] = data->pRgbFloats[srcOff + 2];
+            }
+        }
+    }
+
+    // Pack floats into RpptRGB
+    const size_t n = data->pSrcDesc->n;
+    for (size_t i = 0; i < n; ++i) {
+        const size_t off = i * 3;
+        data->pRgb[i].R = data->pRgbFloats[off + 0];
+        data->pRgb[i].G = data->pRgbFloats[off + 1];
+        data->pRgb[i].B = data->pRgbFloats[off + 2];
+    }
+
+    return status;
+}
+
+static vx_status VX_CALLBACK validateColorCast(vx_node node, const vx_reference parameters[], vx_uint32 num, vx_meta_format metas[]) {
+    vx_status status = VX_SUCCESS;
+    vx_enum scalar_type;
+    // inputLayout
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[5], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_INT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Parameter: #5 type=%d (must be VX_TYPE_INT32)\n", scalar_type);
+    // outputLayout
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[6], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_INT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Parameter: #6 type=%d (must be VX_TYPE_INT32)\n", scalar_type);
+    // roiType
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[7], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_INT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Parameter: #7 type=%d (must be VX_TYPE_INT32)\n", scalar_type);
+    // deviceType
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[8], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_UINT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Parameter: #8 type=%d (must be VX_TYPE_UINT32)\n", scalar_type);
+
+    // Check for input tensor
+    size_t num_tensor_dims;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &num_tensor_dims, sizeof(num_tensor_dims)));
+    if (num_tensor_dims < 4)
+        return ERRMSG(VX_ERROR_INVALID_DIMENSION, "validate: ColorCast: tensor: #0 dimensions=%lu (must be >= 4)\n", num_tensor_dims);
+
+    // Check for output tensor
+    vx_uint8 tensor_fixed_point_position;
+    size_t tensor_dims[RPP_MAX_TENSOR_DIMS];
+    vx_enum tensor_dtype;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_NUMBER_OF_DIMS, &num_tensor_dims, sizeof(num_tensor_dims)));
+    if (num_tensor_dims < 4)
+        return ERRMSG(VX_ERROR_INVALID_DIMENSION, "validate: ColorCast: tensor: #2 dimensions=%lu (must be >= 4)\n", num_tensor_dims);
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DIMS, &tensor_dims, sizeof(tensor_dims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DATA_TYPE, &tensor_dtype, sizeof(tensor_dtype)));
```

Comment:
Do we usually add a check for the data type? I thought we did. Can you check with older kernels or nn kernels and add everywhere if we did before?